manifests/ca.pp

```diff
@@ -175,6 +180,15 @@
     require  => Exec["initca ${name}"],
   }
 
+  if $tls_auth {
+    exec { "generate tls key for ${name}":
+      command  => "openvpn --genkey --secret keys/ta.key",
+      cwd      => "/etc/openvpn/${name}/easy-rsa",
+      creates  => "/etc/openvpn/${name}/easy-rsa/keys/ta.key",
+      provider => 'shell',
+      require => Exec["copy easy-rsa to openvpn config folder ${name}"],
```

Comment:
Shouldn't this be run after `Exec["initca ${name}"]` or even after `Exec["generate server cert ${name}"]`?